### PR TITLE
Add screenlock check and fix tty not appear after booting snapshot

### PR DESCRIPTION
**Pull request by mitiao**
Add screenlock check after zypper patch to fix following issue:
https://openqa.suse.de/tests/407214/modules/yast2_migration/steps/7

Fix tty not appear after booting snapshot
after booting snapshot, tty should appear like this test:
https://openqa.suse.de/tests/394644/modules/snapper_rollback/steps/1

but tty often not appear after booting snapshot like following tests: 
https://openqa.suse.de/tests/399597/modules/snapper_rollback/steps/1
https://openqa.suse.de/tests/407209/modules/snapper_rollback/steps/1

in this case manually switch to tty1 would be successful:
https://openqa.suse.de/tests/399597/modules/consoletest_setup/steps/2

also the message on failure screens are various and hard to needle, do not assert them and check correct linux-login screen only.

### CODE REVIEW

File: tests/online_migration/sle12_online_migration/snapper_rollback.pm [hunk 1]
```diff
@@ -26,10 +26,22 @@ sub check_rollback_system() {
 sub run() {
     my $self = shift;
 
-    # login to snapshot and perform snapper rollback
-    assert_screen 'linux-login', 200;
+    # login to before online migration snapshot
+    # tty would not appear quite often after booting snapshot
```
Comment (okurz) [inline tests/online_migration/sle12_online_migration/snapper_rollback.pm]: why shouldn't it appear? Is this a potential product issue? IMHO the tests should not blindly hit keys up to six times if nothing happens. At least call `record_soft_failure` with a reference to a bug.

File: tests/online_migration/sle12_online_migration/snapper_rollback.pm [hunk 2]
```diff
@@ -26,10 +26,22 @@ sub check_rollback_system() {
...
+            if (check_screen("tty1-selected", 5)) {
+                last;
+            }
+        }
+    }
+
     select_console 'root-console';
     wait_still_screen;
+
+    # perform snapper rollback
```
Comment (okurz) [inline tests/online_migration/sle12_online_migration/snapper_rollback.pm]: stating the obvious. you don't need this comment. If you do, make it another commit as the change is unrelated.

File: tests/online_migration/sle12_online_migration/yast2_migration.pm [hunk 3]
```diff
@@ -19,6 +19,19 @@ sub run {
...
+        send_key "backspace";
+        if (check_screen('screenlock')) {
+            send_key "esc";
+            if (check_screen 'gnome-screenlock-password') {
+                type_password;
+                send_key "ret";
+            }
+        }
+        mouse_hide(1);
+        assert_screen 'generic-desktop';
```
Comment (okurz) [inline tests/online_migration/sle12_online_migration/yast2_migration.pm]: you are duplicating content from `lib/utils.pm::wait_boot` and `tests/installation/first_boot.pm`.

[DRY](https://en.wikipedia.org/wiki/Don't_repeat_yourself). You are calling the same code multiple times, please [extract method](http://refactoring.com/catalog/extractMethod.html).

File: tests/online_migration/sle12_online_migration/snapper_rollback.pm (hunk 1 above)
Comment (dzedro) [inline tests/online_migration/sle12_online_migration/snapper_rollback.pm]: It does not appear because of [bsc#980337](https://bugzilla.suse.com/show_bug.cgi?id=980337) I guess?

**Comment by dzedro** [comment]
Yes it would be successful with selecting tty1, but then we ignore bug [bsc#980337](https://bugzilla.suse.com/show_bug.cgi?id=980337)

Should look like this, expect login prompt.
https://openqa.suse.de/tests/403540/modules/snapper_rollback/steps/1

**Comment by mitiao** [comment]
yes, that should be this bug, and will record soft failure here.
thanks.

File: tests/online_migration/sle12_online_migration/yast2_migration.pm (hunk 3 above)
Comment (mitiao) [inline tests/online_migration/sle12_online_migration/yast2_migration.pm]: utils::wait_boot and first_boot do not contain and not check screenlock.
this reduced snippet came from tests/console/consoletest_finish.pm

Comment (coolo) [inline tests/online_migration/sle12_online_migration/yast2_migration.pm]: doesn't invalidate the DRY argument

Comment (mitiao) [inline tests/online_migration/sle12_online_migration/yast2_migration.pm]: yes, i am trying to extract a method to check screenlock

Comment (mitiao) [inline tests/online_migration/sle12_online_migration/yast2_migration.pm]: like now?

**Comment by okurz** [comment]
LGTM

**Comment by mitiao** [comment]
merge conflict fixed

**Comment by okurz** [comment]
@dzedro second opinion, plz. merge if ok

**Comment by mitiao** [comment]
@dzedro any opinion?
